src/tox.rs

```diff
@@ -12,9 +13,37 @@ const BOOTSTRAP_KEY: &'static str =
 
 const CLIENT_NAME: &'static str = "ws-client";
 
+#[derive(Clone)]
+pub struct ToxGuard {
+    is_dropped: Arc<atomic::AtomicBool>
+}
+
+impl ToxGuard {
```

Comment:
Best practice: oneshot channel

Reply:
https://subscription.packtpub.com/book/application_development/9781788623926/8/ch08lvl1sec75/using-the-oneshot-channel

Reply:
It doesn't have that nice RAII semantics.

Reply:
Still you call drop explicitly https://github.com/tox-rs/ws-tox/pull/22/files#diff-639fbc4ef05b315af92b4d836c31b023R98. You can call .send in that closure too.

Also I think https://docs.rs/futures/0.1.11/src/futures/sync/oneshot.rs.html#272 receiver is marked as finished if all of Senders are dropped.

Reply:
I still don't like to use channel machinery for something as simple as a drop switch.

Reply:
Fine.